go/store/datas/pull/pull_chunk_tracker.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pull
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/dolthub/dolt/go/store/hash"
+)
+
```

Comment:
Can you add a comment for future readers about what all this is for. Something to the effect of enabling parallel calls to the sink DB for HasMany while pulling other chunks.

Reply:
Done. Thanks!